block-producer/src/txqueue/mod.rs

```diff
@@ -158,10 +161,13 @@ where
     TV: TransactionVerifier,
     BB: BatchBuilder,
 {
+    #[allow(clippy::blocks_in_conditions)] // Workaround of `instrument` issue
+    #[instrument(skip_all, err(Debug), fields(COMPONENT = crate::COMPONENT))]
     async fn add_transaction(
         &self,
         tx: SharedProvenTx,
     ) -> Result<(), AddTransactionError> {
+        info!(tx_id = %tx.id().inner(), COMPONENT);
```

Comment:
Can we not extract `tx.id()` as a part of `instrument` on line 165 above? Then we won't need this extra line here.

Reply:
Done. But we should aware of leaking fields from `span`s or `instrument`s to logs of all subsequent calls.